integrationTests/vm/arwen/upgrades/upgrades_test.go

```diff
@@ -0,0 +1,134 @@
+	require.Nil(t, err)
+	require.Equal(t, uint64(24), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+}
+
+func TestUpgrades_HelloCannotBeUpgradedByNonOwner(t *testing.T) {
+	context := arwen.SetupTestContext(t)
+	defer context.Close()
+
+	fmt.Println("Deploy v1")
+
+	context.ScCodeMetadata.Upgradeable = true
+	err := context.DeploySC("../testdata/hello-v1/answer.wasm", "")
+	require.Nil(t, err)
+	require.Equal(t, uint64(24), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+
+	fmt.Println("Upgrade to v2 will not be performed")
+
+	// Alice states that she is the owner of the contract (though she is not)
+	context.Owner = context.Alice
+	err = context.UpgradeSC("../testdata/hello-v2/answer.wasm", "")
+	require.Nil(t, err)
+	require.Equal(t, uint64(24), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+}
+
+func TestUpgrades_HelloUpgradesToNotUpgradeable(t *testing.T) {
+	context := arwen.SetupTestContext(t)
+	defer context.Close()
+
+	fmt.Println("Deploy v1")
+
+	context.ScCodeMetadata.Upgradeable = true
+	err := context.DeploySC("../testdata/hello-v1/answer.wasm", "")
+	require.Nil(t, err)
+	require.Equal(t, uint64(24), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+
+	fmt.Println("Upgrade to v2, becomes not upgradeable")
+
+	context.ScCodeMetadata.Upgradeable = false
+	err = context.UpgradeSC("../testdata/hello-v2/answer.wasm", "")
+	require.Nil(t, err)
+	require.Equal(t, uint64(42), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+
+	fmt.Println("Upgrade to v3, should not be possible")
+
+	err = context.UpgradeSC("../testdata/hello-v3/answer.wasm", "")
+	require.Nil(t, err)
+	require.Equal(t, uint64(42), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+}
+
+func TestUpgrades_ParentAndChildContracts(t *testing.T) {
+	context := arwen.SetupTestContext(t)
+	defer context.Close()
+
+	var parentAddress []byte
+	var childAddress []byte
+	owner := &context.Owner
+
+	fmt.Println("Deploy parent")
+
+	err := context.DeploySC("../testdata/upgrades-parent/parent.wasm", "")
+	require.Nil(t, err)
+	require.Equal(t, uint64(45), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+	parentAddress = context.ScAddress
+
+	fmt.Println("Deploy child v1")
+
+	childInitialCode := arwen.GetSCCode("../testdata/hello-v1/answer.wasm")
+	err = context.ExecuteSC(owner, "createChild@"+childInitialCode)
+	require.Nil(t, err)
+
+	fmt.Println("Aquire child address, do query")
+
+	childAddress = context.QuerySCBytes("getChildAddress", [][]byte{})
+	context.ScAddress = childAddress
+	require.Equal(t, uint64(24), context.QuerySCInt("getUltimateAnswer", [][]byte{}))
+
+	fmt.Println("Deploy child v2")
+	context.ScAddress = parentAddress
+	// We need to double hex-encode the code (so that we don't have to hex-encode in the contract).
+	childUpgradedCode := arwen.GetSCCode("../testdata/hello-v2/answer.wasm")
+	childUpgradedCode = hex.EncodeToString([]byte(childUpgradedCode))
+	// Not supported at this moment, expect error.
```

Comment:
why not supported ?

Reply:
Support for upgrade from indirect call will be done in [EN-6058](https://elrondnetwork.atlassian.net/browse/EN-6058)